### PR TITLE
fix: clean errors, add validation & reduce exposed

### PR DESCRIPTION
**Pull request by a335926a**
This PR fixes a few issues listed below:

- changes errors to be more generic, removing extraneous information to user
- adds validation for urls to only originate from 'http' or 'https' protocols
- sorts requires by alphabetical (node, node_modules, project) order
- only exposes `/_ah/stop` if `NODE_ENV` is set to `development`
- only exposes `progress-bar.html` used in application

### CODE REVIEW

**Comment by a335926a** [comment]
I signed it!

File: src/main.js [hunk 1]
```diff
@@ -124,19 +131,24 @@ app.get('/screenshot/:url(*)', async(request, response) => {
...
-    response.status(400).send(message);
+    response.status(400).send('Cannot render requested URL');
+    console.error(err);
   }
 });
 
 app.get('/_ah/health', (request, response) => response.send('OK'));
 
-app.get('/_ah/stop', async(request, response) => {
+// only expose chrome disable in development
```
Comment (samuelli) [inline src/main.js]: Let's just remove `/_ah/stop` entirely.

File: test/app-test.js [hunk 2]
```diff
@@ -23,12 +23,20 @@ const express = require('express');
 
 const app = express();
 app.use(express.static(path.resolve(__dirname, 'resources')));
+
+const appInstances = [];
 const testBase = 'http://localhost:1234/';
+process.env.NODE_ENV = 'development';
```
Comment (samuelli) [inline test/app-test.js]: Remove per other comment

File: src/main.js [hunk 3]
```diff
@@ -17,24 +17,30 @@
...
 const uuidv4 = require('uuid/v4');
+const cache = require('./cache');
+const renderer = require('./renderer');
+
+const app = express();
+
+const CONFIG_PATH = path.resolve(__dirname, '../config.json');
+const PROGRESS_BAR_PATH = path.resolve(__dirname, '../node_modules/progress-bar-element/progress-bar.html');
+const PORT = process.env.PORT || '3000';
+const ENVIRONMENT = process.env.NODE_ENV;
```
Comment (samuelli) [inline src/main.js]: Remove per below comment.

File: test/app-test.js [hunk 4]
```diff
@@ -39,8 +47,8 @@ test.before(async(t) => {
 async function createServer(config) {
   delete require.cache[require.resolve('../src/main.js')];
   const app = await require('../src/main.js');
-  if (config)
-    app.setConfig(config);
+  if (config) app.setConfig(config);
```
Comment (samuelli) [inline test/app-test.js]: Can we leave this on a separate line please?

File: test/app-test.js [hunk 5]
```diff
@@ -39,8 +47,8 @@ test.before(async(t) => {
...
+  await appInstances.push(app);
```
Comment (samuelli) [inline test/app-test.js]: `await` isn't necessary

**Comment by a335926a** [comment]
Covered the changes requested in the latest commit

**Pull request merged**